### PR TITLE
In 143275155 people search 2

### PR DESCRIPTION
**Pull request by ThomasOSI**

### CODE REVIEW

File: app/controllers/api/v2/people_search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        dates = params[:search_term].match(/\d{4}-\d{2}-\d{2}/)
+        if dates.nil?
+          nil
+        else
+          dates[0]
+        end
+      end
+
+      def date_of_birth_month_day_year
+        dates = params[:search_term].match(%r{\d{1,2}\/\d{1,2}\/\d{4}})
```
Comment (saraOSI) [inline app/controllers/api/v2/people_search_controller.rb]: Is there a reason we assume that the month/day/year format will use slashes and year/month/day will use hyphens?

File: app/controllers/api/v2/people_search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        if dates.nil?
+          nil
```
Comment (saraOSI) [inline app/controllers/api/v2/people_search_controller.rb]: This conditional can also use an early return to make the code block a bit shorter. eg:
```
return nil if dates.nil?
Date.strptime(dates[0], '%m/%d/%Y').to_s(:db)
```

File: docker-compose.yml [hunk 3]
```diff
@@ -19,6 +19,7 @@ services:
       TEST_ELASTICSEARCH_URL: 'http://test_elasticsearch:9200'
       AUTHENTICATION_URL: 'http://www.example.com/'
       AUTHENTICATION: 'false'
+      SEARCH_URL: 'http://tptsearch'
```
Comment (saraOSI) [inline docker-compose.yml]: Has this new env variable been communicated to the necessary teams?

File: config/initializers/api.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  CONTENT_TYPE = 'application/json'
+
+  def self.connection_settings(connection)
+    connection.response :json, content_type: /\bjson$/
+    connection.adapter Faraday.default_adapter
+    connection
+  end
+
+  def self.people_search_connection(path)
+    @connection ||= Faraday.new(url: ENV.fetch('SEARCH_URL') + path) do |connection|
```
Comment (saraOSI) [inline config/initializers/api.rb]: Do we want this to blow up if we haven't set the `SEARCH_URL` env variable, or should there be a default passed to fetch?

File: spec/config/initializers/api_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+require 'rails_helper'
+
+describe API do
+  before do
+    allow(ENV).to receive(:fetch).with('SEARCH_URL')
+      .and_return('http://search_api_url')
+  end
+
+  describe '.make_api_call' do
+    it 'sends search requests to the API' do
```
Comment (saraOSI) [inline spec/config/initializers/api_spec.rb]: There are two conditionals in `make_api_call` that I don't see tested here. The first is whether or not the Content-Type header is included, and the second is whether or not we set the request body. Having scenarios that hit each of those code branches would provide more well-tested code.

File: spec/requests/api/v1/participant_address_spec.rb [hunk 6]
```diff
@@ -7,9 +7,9 @@
       participant = FactoryGirl.create(
         :participant
       )
-      newAddress = FactoryGirl.build(:address)
+      new_address = FactoryGirl.build(:address)
```
Comment (saraOSI) [inline spec/requests/api/v1/participant_address_spec.rb]: Snake case ftw! Though, in a pinch, k3$ha case will do.

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -105,4 +106,10 @@
   #   # test failures related to randomization by passing the same
   #   # `--seed` value as the one that triggered the failure.
   #   Kernel.srand config.seed
+
+  config.around(:each) do |example|
+    WebMock.disable_net_connect!(allow_localhost: true, allow: %(elasticsearch))
```
Comment (saraOSI) [inline spec/spec_helper.rb]: 👍 Nice workaround!

File: spec/controllers/api/v2/people_search_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+        highlight: highlight
+      }
+    end
+
+    it 'creates a search query for names' do
+      expect(API).to receive(:make_api_call)
+        .with('/api/v1/dora/people/_search', :post, name_search)
+        .and_return(response)
+      person_results = get :index, params: { search_term: 'blah' }
+      expect(JSON.parse(person_results.body)).to match array_including(
```
Comment (saraOSI) [inline spec/controllers/api/v2/people_search_controller_spec.rb]: Is there benefit in testing the response body here? From what I can piece together, we're simply testing the mock data that we've stubbed is being returned. Are we transforming that data in some way that would make testing the controller response valuable?

File: app/controllers/api/v2/people_search_controller.rb (hunk 1 above)
Comment (ThomasOSI) [inline app/controllers/api/v2/people_search_controller.rb]: This is the original logic. Good question but not in the scope of this story.

File: docker-compose.yml (hunk 3 above)
Comment (ThomasOSI) [inline docker-compose.yml]: It will ... (on the list)

File: config/initializers/api.rb (hunk 4 above)
Comment (ThomasOSI) [inline config/initializers/api.rb]: This should blow up.

File: spec/requests/api/v1/participant_address_spec.rb (hunk 6 above)
Comment (ThomasOSI) [inline spec/requests/api/v1/participant_address_spec.rb]: Rubocop pointed this one out so we fixed :-)

File: spec/controllers/api/v2/people_search_controller_spec.rb (hunk 8 above)
Comment (ThomasOSI) [inline spec/controllers/api/v2/people_search_controller_spec.rb]: We really care that we are parsing the search_term and building the search (e.g. name_search as shown in line 67) accordingly.  Updated stories are green and chain-saw is happy.

**Pull request merged**